### PR TITLE
Add a parameter to build_run_command methods that controls OTP usage

### PR DESCRIPTION
**Pull request by VakarisZ**
https://github.com/guardicore/monkey/issues/4201

### CODE REVIEW

**Comment by VakarisZ** [comment]
Most methods for linux and windows are the same. I wonder if it would make sense to use inheritance

File: agentpluginapi/i_windows_agent_command_builder.py [hunk 1]
```diff
@@ -54,11 +54,12 @@ def build_download_command(self, download_options: WindowsDownloadOptions):
         """
 
     @abc.abstractmethod
-    def build_run_command(self, run_options: WindowsRunOptions):
+    def build_run_command(self, run_options: WindowsRunOptions, add_otp=True):
```
Comment (ilija-lazoroski) [inline agentpluginapi/i_windows_agent_command_builder.py:57]: Why don't you add this to `{Windows,Linux}RunOptions`??

**Comment by mssalvatore** [comment]
> Most methods for linux and windows are the same. I wonder if it would make sense to use inheritance

We tried that initially. It ends up causing more problems than it solves.

**Comment by mssalvatore** [comment]
We also need the interface to provide a way to get the portion of the command to set the OTP environment variable.

File: agentpluginapi/i_windows_agent_command_builder.py (hunk 1 above)
Comment (VakarisZ) [reply agentpluginapi/i_windows_agent_command_builder.py:57]: Different concerns, I think. One is command build, another is what the command should do. OTP does not describe what the command should do

Comment (ilija-lazoroski) [reply agentpluginapi/i_windows_agent_command_builder.py:57]: Yes, it does. OTP might be part of the command, so it makes sense if we put it in the options along with other stuff that builds the command.

Comment (VakarisZ) [reply agentpluginapi/i_windows_agent_command_builder.py:57]: I looked into it from the usage perspective. I think `RunOptions` depend on the target, `add_otp` depends on whether this command will be ran through a script or directly. I don't think it's a big deal either way, so I let you decide

**Comment by mssalvatore** [comment]
> > We tried that initially. It ends up causing more problems than it solves.
> 
> Right, because of different options, I presume
> 
> > We also need the interface to provide a way to get the portion of the command to set the OTP environment variable.
> 
> Why? OTP is only relevant to the run command and it's best to set it for the duration of the run command (`OTP=something /tmp/m0nk3y`). It makes sense to have them together.

If it's not included in the run command (because the run command is in the bash script), it still needs to be set in the environment the bash script runs in.

**Comment by mssalvatore** [comment]
One solution might be another utility function (maybe a static method on the command builders??) that accepts an OTP as input and generates the necessary command to set the environment variable. the command builder can use this if the otp variable is set and anyone else can use it to generate just that component if they need it.

**Comment by VakarisZ** [comment]
> If it's not included in the run command (because the run command is in the bash script), it still needs to be set in the environment the bash script runs in.

It will be set in the run command that runs the bash script. 
```
run_options = LinuxRunOptions(
                dropper_execution_mode=DropperExecutionMode.SCRIPT,
                agent_destination_path=f"{FILE_PATH}"
agent_command_builder.build_run_command(run_options)
```

**Comment by mssalvatore** [comment]
> > If it's not included in the run command (because the run command is in the bash script), it still needs to be set in the environment the bash script runs in.
> 
> It will be set in the run command that runs the bash script.
> 
> ```
> run_options = LinuxRunOptions(
>                 dropper_execution_mode=DropperExecutionMode.SCRIPT,
>                 agent_destination_path=f"{FILE_PATH}")
> agent_command_builder.build_run_command(run_options)
> ```

Ok. Got it.

File: agentpluginapi/i_linux_agent_command_builder.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ class LinuxRunOptions(InfectionMonkeyBaseModel):
     agent_destination_path: PurePosixPath
     dropper_execution_mode: DropperExecutionMode
     dropper_destination_path: Optional[PurePosixPath] = None
+    add_otp: bool = True
```
Comment (mssalvatore) [inline agentpluginapi/i_linux_agent_command_builder.py:32]: NIT: Maybe "include_otp" would be a better wording.

**Review by mssalvatore — APPROVED** [review]

File: agentpluginapi/i_windows_agent_command_builder.py (hunk 1 above)
Comment (mssalvatore) [reply agentpluginapi/i_windows_agent_command_builder.py]: I'm still not totally convinced we shouldn't switch to just passing parameters to these methods, but that's not a problem to be solved in this PR.

**Pull request merged**